File: sdks/python/examples/quickstart.py
#!/usr/bin/env python3
# Copyright 2026 hunta.ai
# SPDX-License-Identifier: Apache-2.0
"""Hunta Gather Python SDK quickstart.

  export GATHER_TOKEN=<your-key>
  export GATHER_URL=https://mcp.hunta.ai    # optional; this is the default
  python examples/quickstart.py

Tenancy is carried by the token: no tenant argument, and you cannot request
another tenant's data. An owner key writes canon directly (gather), as shown
here for a one-shot demo. In production, give agents propose-only keys and use
capture / propose_errata so a curator admits into canon.
"""
from hunta_gather import GatherClient

with GatherClient() as gather:
    who = gather.whoami()
    print("whoami:", who)

    # store a memory (owner key: straight to canon)
    stored = gather.gather("Our design partner is Acme Corp.")
    print("gather:", stored)

    # recall it (returns only sealed facts)
    hits = gather.recall("design partner", limit=5)
    print("recall:", hits)

    # propose-only path: stage a candidate for a curator to admit
    candidate = gather.capture("Acme's primary contact is Dana.", source="api")
    print("capture:", candidate)

File: sdks/ts/examples/quickstart.mjs
// Hunta Gather TypeScript SDK quickstart.
//
//   export GATHER_URL=https://mcp.hunta.ai        # or your Estate deployment
//   export GATHER_TOKEN=<your-agent-key>          # propose + recall is enough here
//   node examples/quickstart.mjs
//
// The SDK is governed by construction: gather() PROPOSES a candidate into staging,
// it does not write canon. A curator admits it later. Recall returns only sealed
// facts. Tenancy is the token's tid claim: there is no tenant argument.

import { GatherClient } from "@hunta/gather";

const gather = new GatherClient(); // reads GATHER_TOKEN / GATHER_URL from the env

// propose a memory (staging inbox, never recalled until a curator promotes it)
const proposed = await gather.gather("Our design partner is Acme Corp.");
console.log("gather (proposed candidate):", proposed);

// recall sealed facts
const hits = await gather.recall("design partner", { limit: 5 });
console.log("recall:", hits);

// who am I, and what can this token do
console.log("whoami:", await gather.whoami());

// Curator (needs a memory:curate or memory:write token). Kept separate on purpose:
//   await gather.curate.promoteCandidate(proposed.candidate_id);
